2019-12-03 13:32:35,955 ERROR: Exception on /simple.png [GET] [in /scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/app.py:1891]
Traceback (most recent call last):
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/scratch/knerr/repos/mydrop/mytrilog/app/routes.py", line 221, in simple
    from matplotlib.backends.backend_agg import FigureCanvasAgg as FigureCanvas
ModuleNotFoundError: No module named 'matplotlib'
2019-12-03 13:34:37,682 INFO: MyTriLog startup... [in /scratch/knerr/repos/mydrop/mytrilog/app/__init__.py:49]
2019-12-03 13:34:39,173 ERROR: Exception on /simple.png [GET] [in /scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/app.py:1891]
Traceback (most recent call last):
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/scratch/knerr/repos/mydrop/mytrilog/app/routes.py", line 196, in simple
    from matplotlib.backends.backend_agg import FigureCanvasAgg as FigureCanvas
ModuleNotFoundError: No module named 'matplotlib'
2019-12-03 14:00:22,654 INFO: MyTriLog startup... [in /scratch/knerr/repos/mydrop/mytrilog/app/__init__.py:49]
2019-12-03 14:01:49,020 INFO: MyTriLog startup... [in /scratch/knerr/repos/mydrop/mytrilog/app/__init__.py:49]
2019-12-03 14:02:10,282 INFO: MyTriLog startup... [in /scratch/knerr/repos/mydrop/mytrilog/app/__init__.py:49]
2019-12-03 14:03:21,454 INFO: MyTriLog startup... [in /scratch/knerr/repos/mydrop/mytrilog/app/__init__.py:49]
2019-12-03 14:07:07,068 INFO: MyTriLog startup... [in /scratch/knerr/repos/mydrop/mytrilog/app/__init__.py:49]
2019-12-03 14:16:11,352 INFO: MyTriLog startup... [in /scratch/knerr/repos/mydrop/mytrilog/app/__init__.py:49]
2019-12-03 14:18:48,270 INFO: MyTriLog startup... [in /scratch/knerr/repos/mydrop/mytrilog/app/__init__.py:49]
2019-12-03 14:21:18,775 INFO: MyTriLog startup... [in /scratch/knerr/repos/mydrop/mytrilog/app/__init__.py:49]
2019-12-03 14:31:38,898 INFO: MyTriLog startup... [in /scratch/knerr/repos/mydrop/mytrilog/app/__init__.py:49]
2019-12-03 14:31:42,787 ERROR: Exception on /index [GET] [in /scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/app.py:1891]
Traceback (most recent call last):
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/scratch/knerr/repos/mydrop/mytrilog/app/routes.py", line 42, in index
    prev_url=prev_url)
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/templating.py", line 140, in render_template
    ctx.app,
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask/templating.py", line 120, in _render
    rv = template.render(context)
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/scratch/knerr/repos/mydrop/mytrilog/app/templates/index.html", line 3, in top-level template code
    {% import 'bootstrap/wtf.html' as wtf %}
  File "/scratch/knerr/repos/mydrop/mytrilog/app/templates/base.html", line 1, in top-level template code
    {% extends 'bootstrap/base.html' %}
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 1, in top-level template code
    {% block doc -%}
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 4, in block "doc"
    {%- block html %}
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 20, in block "html"
    {% block body -%}
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 23, in block "body"
    {% block content -%}
  File "/scratch/knerr/repos/mydrop/mytrilog/app/templates/base.html", line 46, in block "content"
    {% block app_content %}{% endblock %}
  File "/scratch/knerr/repos/mydrop/mytrilog/app/templates/index.html", line 27, in block "app_content"
    {% include '_workout-short.html' %}
  File "/scratch/knerr/repos/mydrop/mytrilog/app/templates/_workout-short.html", line 5, in top-level template code
    <td>{{db.User.query.get(workout.who)}}</td>
  File "/scratch/knerr/venvs/mytrilog/venv/lib/python3.6/site-packages/jinja2/environment.py", line 430, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'db' is undefined
2019-12-03 14:33:26,463 INFO: MyTriLog startup... [in /scratch/knerr/repos/mydrop/mytrilog/app/__init__.py:49]
2019-12-03 14:50:52,280 INFO: MyTriLog startup... [in /scratch/knerr/repos/mydrop/mytrilog/app/__init__.py:49]
2019-12-03 14:51:50,349 INFO: MyTriLog startup... [in /scratch/knerr/repos/mydrop/mytrilog/app/__init__.py:49]
2019-12-03 14:54:59,029 INFO: MyTriLog startup... [in /scratch/knerr/repos/mydrop/mytrilog/app/__init__.py:49]
2019-12-03 14:55:54,223 INFO: MyTriLog startup... [in /scratch/knerr/repos/mydrop/mytrilog/app/__init__.py:49]
2019-12-03 14:56:59,691 INFO: MyTriLog startup... [in /scratch/knerr/repos/mydrop/mytrilog/app/__init__.py:49]
